server/src/main/java/org/candlepin/auth/KeycloakAuth.java

```diff
@@ -72,23 +90,20 @@ public KeycloakAuth(UserServiceAdapter userServiceAdapter, Provider<I18n> i18nPr
     }
 
     private void createKeycloakDeploymentFrom() {
-        try {
 
+        try {
             InputStream is = loadKeycloakConfigFile();
             kd = KeycloakDeploymentBuilder.build(is);
             deploymentContext = new AdapterDeploymentContext(kd);
-
         }
-        catch (Exception e) {
-
+        catch (FileNotFoundException e) {
+            log.error("{} :File not found", filepath);
         }
-
     }
 
     private InputStream loadKeycloakConfigFile() throws FileNotFoundException {
-        String filepath = "/etc/candlepin/keycloak.json";
+        filepath = ConfigProperties.KEYCLOAK_FILEPATH;
         return new FileInputStream(filepath);
```

Comment:
Looking at this whole thing in depth, a few of things come to mind:

1. The keycloak file in question is a configuration file. Candlepin already has its own configuration file and format, and we could just as easily store everything there instead of introducing another file dependency.
2. If we want to stay with a secondary configuration file for this, we should still be loading it here in a way that lets us modify it as we need to, setting reasonable defaults and whatnot as necessary. Following this, I would argue passing around blackbox, unvalidated input streams is not the way to go.
3. This is still a hard-coded path, not a configurable one. 

In any case, we should be building the AdapterConfig ourselves and passing that in to KeycloakdeploymentBuilder.build, rather than throwing a file at it and hoping for the best. Additionally, I would argue the relevant configurations should be stored in CP's config rather than a separate config entirely.

To recap: At the absolute least, this needs to *not* be a hardcoded constant file. This should be a configurable reference, or values from the CP configuration, used to build an AdapterConfig that is passed into the KeycloakDeploymentBuilder.

Reply:
> 1. The keycloak file in question is a configuration file. Candlepin already has its own configuration file and format, and we could just as easily store everything there instead of introducing another file dependency.

The idiomatic way to integrate with Keycloak is to generate keycloak.json through the Keycloak admin interface, and then land this as part of the configuration of your application. In order to keep things simple and idiomatic, I want to ask that we keep keycloak.json separate from the rest of candlepin configuration.

> 2. If we want to stay with a secondary configuration file for this, we should still be loading it here in a way that lets us modify it as we need to, setting reasonable defaults and whatnot as necessary. Following this, I would argue passing around blackbox, unvalidated input streams is not the way to go.

Agree, it looks like we can use [`KeycloakDeployBuilder.loadAdapterConfig(InputStream)`](https://github.com/keycloak/keycloak/blob/ccc8e06f9a7d11db9c2848f5891bdb3f75c40e33/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/KeycloakDeploymentBuilder.java#L191), and then apply any overrides to the loaded AdapterConfig object. I don't think we have any overrides we need to make at the moment, so @ShwethaKalyanaraman please leave a comment that leaves a hint where we can apply overrides. something like:

```java
AdapterConfig config = KeycloakDeployBuilder.loadAdapterConfig(inputStream);
// NOTE: Candlepin-specific overrides to keycloak config can be made in this method.
return config;
```

> 3. This is still a hard-coded path, not a configurable one.

I'll suggest a name: `candlepin.keycloak.config_path`, with a default of `/etc/candlepin/keycloak.json`.